File: CS/ShowMessageAfterCallback/Controllers/HomeController.cs
using System;
using System.Web.Mvc;

namespace ShowMessageAfterCallback.Controllers {
    public class HomeController: Controller {
        public const string EditResultKey = "EditResult";
        public const string EditErrorKey = "EditError";

        Repository repository = new Repository();

        public ActionResult Index() {
            return View();
        }
        public ActionResult GridViewPartial() {
            return PartialView(repository.GetItems());
        }
        public ActionResult GridViewPartialAddNew(DataEntry entry) {
            if (ModelState.IsValid) {
                try {
                    repository.Add(entry);
                    ViewData[EditResultKey] = string.Format("ADDED WITH KEY: '{0}'", entry.ID);
                } catch (Exception e) {
                    ViewData[EditErrorKey] = e.Message;
                }
            } else
                ViewData[EditErrorKey] = "Please, correct all errors.";
            return PartialView("GridViewPartial", repository.GetItems());
        }
        public ActionResult GridViewPartialUpdate(DataEntry entry) {
            if (ModelState.IsValid) {
                try {
                    repository.Update(entry);
                    ViewData[EditResultKey] = string.Format("UPDATED WITH KEY: '{0}'", entry.ID);
                } catch (Exception e) {
                    ViewData[EditErrorKey] = e.Message;
                }
            } else
                ViewData[EditErrorKey] = "Please, correct all errors.";
            return PartialView("GridViewPartial", repository.GetItems());
        }
        public ActionResult GridViewPartialDelete(int id) {
            if (id >= 0) {
                try {
                    repository.Delete(id);
                    ViewData[EditResultKey] = string.Format("DELETED WITH KEY: '{0}'", id);
                } catch (Exception e) {
                    ViewData[EditErrorKey] = e.Message;
                }
            }
            return PartialView("GridViewPartial", repository.GetItems());
        }
    }
}

File: VB/ShowMessageAfterCallback/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc

Namespace ShowMessageAfterCallback.Controllers
	Public Class HomeController
		Inherits Controller

		Public Const EditResultKey As String = "EditResult"
		Public Const EditErrorKey As String = "EditError"

		Private repository As New Repository()

		Public Function Index() As ActionResult
			Return View()
		End Function
		Public Function GridViewPartial() As ActionResult
			Return PartialView(repository.GetItems())
		End Function
		Public Function GridViewPartialAddNew(ByVal entry As DataEntry) As ActionResult
			If ModelState.IsValid Then
				Try
					repository.Add(entry)
					ViewData(EditResultKey) = String.Format("ADDED WITH KEY: '{0}'", entry.ID)
				Catch e As Exception
					ViewData(EditErrorKey) = e.Message
				End Try
			Else
				ViewData(EditErrorKey) = "Please, correct all errors."
			End If
			Return PartialView("GridViewPartial", repository.GetItems())
		End Function
		Public Function GridViewPartialUpdate(ByVal entry As DataEntry) As ActionResult
			If ModelState.IsValid Then
				Try
					repository.Update(entry)
					ViewData(EditResultKey) = String.Format("UPDATED WITH KEY: '{0}'", entry.ID)
				Catch e As Exception
					ViewData(EditErrorKey) = e.Message
				End Try
			Else
				ViewData(EditErrorKey) = "Please, correct all errors."
			End If
			Return PartialView("GridViewPartial", repository.GetItems())
		End Function
		Public Function GridViewPartialDelete(ByVal id As Integer) As ActionResult
			If id >= 0 Then
				Try
					repository.Delete(id)
					ViewData(EditResultKey) = String.Format("DELETED WITH KEY: '{0}'", id)
				Catch e As Exception
					ViewData(EditErrorKey) = e.Message
				End Try
			End If
			Return PartialView("GridViewPartial", repository.GetItems())
		End Function
	End Class
End Namespace